### ISSUE TITLE
Composer lock file updating not enabled in hosted app

### ISSUE THREAD

**Issue by felixfbecker** [renovatebot/renovate#2465]
**What Renovate type are you using?**
Renovate GitHub App

**Describe the bug**
I got a PR to pin dependencies, but it only updated composer.json, it is actually invalid now and out-of-sync with composer.lock (the version in composer.lock is behind composer.json)

**To Reproduce**

1. Depend on an outdated version with Composer
2. Set up Renovate
3. Get pin PR

**Expected behavior**
It should update the lockfile too (as composer would locally)

**Additional context**
https://github.com/felixfbecker/vscode-php-intellisense/pull/351/files#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780R5

**Comment by rarkins** [comment]
Caused by 
https://github.com/renovatebot/renovate/issues/1357#issuecomment-410938871

**Comment by rarkins** [comment]
Composer lockfile updating is now live in the hosted app

**Comment by rarkins** [comment]
For self-hosted, the best approach is to have `composer` installed globally and Renovate should then call out to it using child_process. If you already created a PR *without* `composer.lock` then it won't get retried non-stop though. You will need to tell Renovate to force rebase it by adding the prefix "rebase!" to the PR title. Please try that with `LOG_LEVEL=debug` set and if it doesn't work then raise an issue in the renovatebot/config-help repo including the logs.

**Comment by evilebottnawi** [comment]
@rarkins i create `renovate.json` with content:
```json
{
  "extends": ["config:base", ":maintainLockFilesWeekly"],
  "composer": {
    "enabled": true
  }
}
```

And renovate will start create PR, but without lock file. We have `composer.lock` in repo. Rebase doesn't solve problem. Debug:
```shell
DEBUG: GET (repository=myorg/myrepo)
       "path": "projects/myorg%2myrepo/repository/files/composer.json?ref=master"
DEBUG: composer.extractDependencies() (repository=myorg/myrepo)
DEBUG: getFile(filePath=composer.lock, branchName=undefined) (repository=myorg/myrepo)
DEBUG: GET (repository=myorg/myrepo)
       "path": "projects/myorg%2Fmyrepo/repository/files/composer.lock?ref=master"
DEBUG: Found composer lock file (repository=myorg/myrepo, packageFile=composer.json)
```

But no lock file in PR. We use gitlab.

**Comment by rarkins** [comment]
That’s not the part of the log I need. There’s a part where Renovate attempts to update composer.lock after patching composer.json. Maybe search the entire log for “composer” and “lockfile” case insensitive.

**Comment by evilebottnawi** [comment]
@rarkins only `npm` lock file output exists :confused: Maybe i can send log file on email?

**Comment by evilebottnawi** [comment]
@rarkins maybe help
```shell
DEBUG: Checking if branch exists: renovate/phpunit-php-code-coverage-6.x (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: branchExists=true (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Branch has 1 upgrade(s) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: recreateClosed is false (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: findPr(renovate/phpunit-php-code-coverage-6.x, chore(deps): update dependency phpunit/php-code-coverage to v6, !open) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: prAlreadyExisted=false (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Checking if PR has been edited (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: getBranchPr(renovate/phpunit-php-code-coverage-6.x) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Checking if branch exists: renovate/phpunit-php-code-coverage-6.x (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Got res with 7 results (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: getPr(2217) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: GET (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
"path": "projects/myorg%2Fmyrepo/merge_requests/2217"
DEBUG: Found existing branch PR (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Checking schedule(, null) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: No schedule defined (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Checking if branch exists: renovate/phpunit-php-code-coverage-6.x (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Branch already exists (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: getBranchPr(renovate/phpunit-php-code-coverage-6.x) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Checking if branch exists: renovate/phpunit-php-code-coverage-6.x (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Got res with 7 results (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: getPr(2217) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Branch does not need rebasing (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Using parentBranch: renovate/phpunit-php-code-coverage-6.x (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: manager.getUpdatedPackageFiles() (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: getFile(filePath=composer.json, branchName=renovate/phpunit-php-code-coverage-6.x) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: GET (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
"path": "projects/myorg%2Fmyrepo/repository/files/composer.json?ref=renovate/phpunit-php-code-coverage-6.x"
DEBUG: npm.updateDependency(): require-dev.phpunit/php-code-coverage = ~6.0.0 (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Version is already updated (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: bumpVersion() (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: No package files need updating (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Getting updated lock files (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: lock file dirs (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
"dirs": {
"yarnLockDirs": [],
"npmLockDirs": [],
"pnpmShrinkwrapDirs": [],
"lernaDirs": []
}
DEBUG: getFile(filePath=lerna.json, branchName=undefined) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Writing package.json files (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
"packageFiles": ["package.json"]
DEBUG: getFile(filePath=package.json, branchName=undefined) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Writing package-lock.json (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: getFile(filePath=package-lock.json, branchName=undefined) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Writing any updated package files (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: No updated lock files in branch (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: No files to commit (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Checking if we can automerge branch (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: mergeStatus=no automerge (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Ensuring PR (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: There are 0 errors and 0 warnings (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: getBranchPr(renovate/phpunit-php-code-coverage-6.x) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Checking if branch exists: renovate/phpunit-php-code-coverage-6.x (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Got res with 7 results (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: getPr(2217) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Found existing PR (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: getChangeLogJSON(args) (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
"args": {
"manager": "composer",
"versionScheme": "semverComposer",
"depType": "require-dev",
"depName": "phpunit/php-code-coverage",
"fromVersion": "5.3.2",
"toVersion": "6.0.7",
"repositoryUrl": "https://github.com/sebastianbergmann/php-code-coverage"
}
DEBUG: Processing existing PR (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Merge Request #2217 does not need updating (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: Checking #2217 for automerge (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
"automerge": false,
"automergeType": "pr",
"automergeComment": "automergeComment"
DEBUG: No automerge (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: processBranch with 1 upgrades (repository=myorg/myrepo, dependencies=phpunit/php-code-coverage, branch=renovate/phpunit-php-code-coverage-6.x)
DEBUG: processBranch() (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Checking if branch exists: renovate/phpunit-phpunit-7.x (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: branchExists=true (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Branch has 1 upgrade(s) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: recreateClosed is false (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: findPr(renovate/phpunit-phpunit-7.x, chore(deps): update dependency phpunit/phpunit to v7, !open) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: prAlreadyExisted=false (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Checking if PR has been edited (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: getBranchPr(renovate/phpunit-phpunit-7.x) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Checking if branch exists: renovate/phpunit-phpunit-7.x (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Got res with 7 results (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: getPr(2218) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: GET (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
"path": "projects/myorg%2Fmyrepo/merge_requests/2218"
DEBUG: Found existing branch PR (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Checking schedule(, null) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: No schedule defined (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Checking if branch exists: renovate/phpunit-phpunit-7.x (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Branch already exists (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: getBranchPr(renovate/phpunit-phpunit-7.x) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Checking if branch exists: renovate/phpunit-phpunit-7.x (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Got res with 7 results (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: getPr(2218) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Branch does not need rebasing (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Using parentBranch: renovate/phpunit-phpunit-7.x (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: manager.getUpdatedPackageFiles() (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: getFile(filePath=composer.json, branchName=renovate/phpunit-phpunit-7.x) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: GET (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
"path": "projects/myorg%2Fmyrepo/repository/files/composer.json?ref=renovate/phpunit-phpunit-7.x"
DEBUG: npm.updateDependency(): require-dev.phpunit/phpunit = ~7.3.0 (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Version is already updated (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: bumpVersion() (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: No package files need updating (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Getting updated lock files (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: lock file dirs (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
"dirs": {
"yarnLockDirs": [],
"npmLockDirs": [],
"pnpmShrinkwrapDirs": [],
"lernaDirs": []
}
DEBUG: getFile(filePath=lerna.json, branchName=undefined) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Writing package.json files (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
"packageFiles": ["package.json"]
DEBUG: getFile(filePath=package.json, branchName=undefined) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Writing package-lock.json (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: getFile(filePath=package-lock.json, branchName=undefined) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Writing any updated package files (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: No updated lock files in branch (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: No files to commit (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Checking if we can automerge branch (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: mergeStatus=no automerge (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Ensuring PR (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: There are 0 errors and 0 warnings (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: getBranchPr(renovate/phpunit-phpunit-7.x) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Checking if branch exists: renovate/phpunit-phpunit-7.x (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Branch exists (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Got res with 7 results (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: getPr(2218) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Found existing PR (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: getChangeLogJSON(args) (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
"args": {
"manager": "composer",
"versionScheme": "semverComposer",
"depType": "require-dev",
"depName": "phpunit/phpunit",
"fromVersion": "6.5.13",
"toVersion": "7.3.5",
"repositoryUrl": "https://github.com/sebastianbergmann/phpunit"
}
DEBUG: Processing existing PR (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Merge Request #2218 does not need updating (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: Checking #2218 for automerge (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
"automerge": false,
"automergeType": "pr",
"automergeComment": "automergeComment"
DEBUG: No automerge (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
DEBUG: processBranch with 1 upgrades (repository=myorg/myrepo, dependencies=phpunit/phpunit, branch=renovate/phpunit-phpunit-7.x)
```

**Comment by rarkins** [comment]
@evilebottnawi you'll only see Renovate attempt `composer.lock` updating *if* it detects there is a change in `composer.json` in the same run. In the log you pasted, both the PRs look to be existing and not in need of an update.

I just verified this functionality myself and it works perfectly fine. 

```
❯ renovate --version
13.85.1

/tmp/composerlock2
❯ composer --version
Composer version 1.6.5 2018-05-04 11:44:59

/tmp/composerlock2
❯ renovate renovate-tests/composerlock2 --log-level=debug > composerlock2.log
```

The above worked perfectly. Result: https://github.com/renovate-tests/composerlock2/pulls

Example logs:
```
DEBUG: Updating packageFile content (repository=renovate-tests/composerlock2, dependencies=monolog/monolog, branch=renovate/monolog-monolog-1.x)
DEBUG: composer.getLockFile(composer.json) (repository=renovate-tests/composerlock2, dependencies=monolog/monolog, branch=renovate/monolog-monolog-1.x)
DEBUG: Using composer cache /var/folders/1y/nbqcg2hn73dg7bc9vpkbn18r0000gn/T/renovate/cache/composer (repository=renovate-tests/composerlock2, dependencies=monolog/monolog, branch=renovate/monolog-monolog-1.x)
 INFO: Running composer via global composer (repository=renovate-tests/composerlock2, dependencies=monolog/monolog, branch=renovate/monolog-monolog-1.x)
DEBUG: composer update command (repository=renovate-tests/composerlock2, dependencies=monolog/monolog, branch=renovate/monolog-monolog-1.x)
       "cmd": "composer",
       "args": "update monolog/monolog --ignore-platform-reqs --no-ansi"
 INFO: Generated lockfile (repository=renovate-tests/composerlock2, dependencies=monolog/monolog, branch=renovate/monolog-monolog-1.x)
       "seconds": 14,
       "type": "composer.lock",
       "stdout": "",
```

In short: Rename and close at least one of those existing PRs if you want to give Renovate a chance to retry creating it with lockfile.

Please raise a config-help issue as here is not the appropriate place to be discussing a different problem. I won't be responding here again as it's too easy to miss updates. I didn't even see your last logs in my notifications until I came here and looked up the closed issue manually.
